### PR TITLE
EN-19215: Skip grouping when querying the rollup if it already matches the query grouping.

### PR DESCRIPTION
**Pull request by znep**
In order to allow using rollups to speed up common grouped queries, we want to be able to skip
doing the grouping if the rollup is already grouped the exact way the query wants.  This doesn't
matter for the smaller rollups that the code was originally designed to support but for other
uses that are large after grouping it makes a big difference.

### CODE REVIEW

File: query-coordinator/src/main/scala/com/socrata/querycoordinator/QueryRewriter.scala [hunk 1]
```diff
@@ -366,16 +394,43 @@ class QueryRewriter(analyzer: SoQLAnalyzer[SoQLType]) {
...
+       * query: SELECT crime_type, count(*), max(severity) GROUP BY crime_type
+       * previous rollup query: SELECT c1, sum(c2), max(c3) GROUP BY c1
+       * desired rollup query:  SELECT c1, c2, c3
+       */
+      val shouldRemoveAggregates = groupBy match {
+        case Some(None) => q.groupBy.isDefined
+        case _ => false
+      }
+
+      val selection = rewriteSelection(q.selection, r.selection, rollupColIdx) match {
```
Comment (znep) [inline query-coordinator/src/main/scala/com/socrata/querycoordinator/QueryRewriter.scala]: Is there a more concise way to express this?

File: query-coordinator/src/main/scala/com/socrata/querycoordinator/QueryRewriter.scala [hunk 2]
```diff
@@ -366,16 +394,43 @@ class QueryRewriter(analyzer: SoQLAnalyzer[SoQLType]) {
...
+        case Some(s) if shouldRemoveAggregates => Some(s.mapValues(removeAggregates))
+        case x => x
+      }
+
```
Comment (chitang) [inline query-coordinator/src/main/scala/com/socrata/querycoordinator/QueryRewriter.scala]: map seems more concise than match

val selection = rewriteSelection(q.selection, r.selection, rollupColIdx) map { s =>
        if (shouldRemoveAggregates) s.mapValues(removeAggregates)
        else s
      }

File: query-coordinator/src/main/scala/com/socrata/querycoordinator/QueryRewriter.scala [hunk 3]
```diff
@@ -366,16 +394,43 @@ class QueryRewriter(analyzer: SoQLAnalyzer[SoQLType]) {
...
+      val orderBy = rewriteOrderBy(q.orderBy, rollupColIdx) match {
```
Comment (chitang) [inline query-coordinator/src/main/scala/com/socrata/querycoordinator/QueryRewriter.scala]: map seems more concise than match again.

File: query-coordinator/src/main/scala/com/socrata/querycoordinator/QueryRewriter.scala (hunk 2 above)
Comment (znep) [reply query-coordinator/src/main/scala/com/socrata/querycoordinator/QueryRewriter.scala]: Agree, fixed.  Thanks!

**Comment by chitang** [comment]
:shipit:

**Pull request merged**